darts/models/forecasting/tsmixer_model.py

```diff
@@ -0,0 +1,943 @@
+"""
+Time-Series Mixer (TSMixer)
+-------
+The inner layers (``nn.Modules``) and the ``TimeBatchNorm2d`` were provided by a PyTorch implementation
+of TSMixer: https://github.com/ditschuk/pytorch-tsmixer
+
+The License of pytorch-tsmixer v0.2.0 from https://github.com/ditschuk/pytorch-tsmixer/blob/main/LICENSE,
+accessed Thursday, March 21st, 2024:
+'The MIT License
+
+Copyright 2023 Konstantin Ditschuneit
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+associated documentation files (the “Software”), to deal in the Software without restriction,
+including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
+subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial
+portions of the Software.
+'
+"""
+
+from __future__ import annotations
+
+import operator
+from collections.abc import Callable
+from functools import reduce
+from typing import Tuple
+
+import torch
+import torch.nn.functional as F
+from torch import nn
+
+from darts.logging import get_logger, raise_log
+from darts.models.components import layer_norm_variants
+from darts.models.forecasting.pl_forecasting_module import (
+    PLMixedCovariatesModule,
+    io_processor,
+)
+from darts.models.forecasting.torch_forecasting_model import MixedCovariatesTorchModel
+from darts.utils.torch import MonteCarloDropout
+
+MixedCovariatesTrainTensorType = Tuple[
+    torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor
+]
+
+logger = get_logger(__name__)
+
+
+def time_to_feature(x: torch.Tensor) -> torch.Tensor:
+    """Converts a time series torch.Tensor to a feature torch.Tensor."""
+    return x.permute(0, 2, 1)
+
+
+feature_to_time = time_to_feature
```

Comment:
why are you doing that?

I have the following questions:

1. why are you not using the method directly?
2. it is confusing not? one is `time_to_feature` and the other `feature_to_time`. They are the opposite, but they are equal
3. Can you not use the name: `time_series_to_feature`, and use this method everywhere?

Reply:
Yes, they both do the same thing.
The reason was to make it more readable in this case:

```
def forward(self, x: torch.Tensor) -> torch.Tensor:
    x_temp = feature_to_time(x) # permute the feature dim with the time dim
    x_temp = self.activation(self.fc1(x_temp))
    x_temp = self.dropout(x_temp)
    x_res = time_to_feature(x_temp) # permute back the time dim with the feature dim
    x_temp = self.norm(x + x_res)
    return x_temp
 ```
    
So a user who looks at the forward pass can better understand what is going on.
But it may be confusing to have the same function twice.

Maybe removing the second function and adding comments where necessary to the forward pass could be a good middle ground. 
I will do that.